srcs/incs/HttpResponseFactory.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef SRCS_INCS_HTTPRESPONSEFACTORY_HPP_
+#define SRCS_INCS_HTTPRESPONSEFACTORY_HPP_
+
+#define CALL_MEMBER_FN(object, ptrToMember) ((object).*(ptrToMember))
+
+#include "IResponseFactory.hpp"
+#include "HttpRequestDTO.hpp"
+#include "HttpGetResponse.hpp"
+#include "HttpPostResponse.hpp"
+#include "HttpDeleteResponse.hpp"
+#include "HttpRedirectionResponse.hpp"
+
+class HttpResponseFactory: public IResponseFactory {
+	public:
+		HttpResponseFactory(const std::string raw_request);
+		~HttpResponseFactory();
+		IResponse *response();
+
+	private:
+		HttpResponseFactory();
+
+		typedef  IResponse *(HttpResponseFactory::* responseCreatorMethod)();
```

Comment:
[Pointer to member function](https://isocpp.org/wiki/faq/pointers-to-members#fnptr-vs-memfnptr-types)